src/Features/Blockcore.Features.WebHost/WebHostFeature.cs

```diff
@@ -0,0 +1,225 @@
+        /// <param name="builder">The string builder to add the settings to.</param>
+        /// <param name="network">The network to base the defaults off.</param>
+        public static void BuildDefaultConfigurationFile(StringBuilder builder, Network network)
+        {
+            WebHostSettings.BuildDefaultConfigurationFile(builder, network);
+        }
+
+        /// <inheritdoc />
+        public override void Dispose()
+        {
+            // Make sure the timer is stopped and disposed.
+            if (this.settings.KeepaliveTimer != null)
+            {
+                this.settings.KeepaliveTimer.Stop();
+                this.settings.KeepaliveTimer.Enabled = false;
+                this.settings.KeepaliveTimer.Dispose();
+            }
+
+            // Make sure we are releasing the listening ip address / port.
+            if (this.webHost != null)
+            {
+                this.logger.LogInformation("API stopping on URL '{0}'.", this.settings.ApiUri);
+                this.webHost.StopAsync(TimeSpan.FromSeconds(WebHostStopTimeoutSeconds)).Wait();
+                this.webHost = null;
+            }
+        }
+    }
+
+    public sealed class WebHostFeatureOptions
+    {
+        public IClientEvent[] EventsToHandle { get; set; }
+
+        public (Type Broadcaster, ClientEventBroadcasterSettings clientEventBroadcasterSettings)[] ClientEventBroadcasters { get; set; }
+    }
+
+    /// <summary>
+    /// A class providing extension methods for <see cref="IFullNodeBuilder"/>.
+    /// </summary>
+    public static class ApiFeatureExtension
+    {
+        [Obsolete("Use the new UseWebHost()")]
+        public static IFullNodeBuilder UseApi(this IFullNodeBuilder fullNodeBuilder, Action<WebHostFeatureOptions> optionsAction = null)
+        {
+            return UseWebHost(fullNodeBuilder, optionsAction);
+        }
+
+        public static IFullNodeBuilder UseWebHost(this IFullNodeBuilder fullNodeBuilder, Action<WebHostFeatureOptions> optionsAction = null)
+        {
+            // TODO: move the options in to the feature builder
+            var options = new WebHostFeatureOptions();
+            optionsAction?.Invoke(options);
+
+            // If there is no events to handle defined in the options configured by the node, add a few default.
+            if (options.EventsToHandle == null)
+            {
+                options.EventsToHandle = new[]
+                {
+                    (IClientEvent) new BlockConnectedClientEvent(),
+                    new TransactionReceivedClientEvent()
+                };
+            }
+
+            // If there is now client event broadcasters, add the default ones.
+            if (options.ClientEventBroadcasters == null)
+            {
+                options.ClientEventBroadcasters = new[]
+                {
+                    (Broadcaster: typeof(StakingBroadcaster), ClientEventBroadcasterSettings: new ClientEventBroadcasterSettings { BroadcastFrequencySeconds = 5 }),
```

Comment:
So we agreed broadcasters will go in their projects right?

Reply:
Yepp, working on it